src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -1954,7 +1954,9 @@ private void CheckMemberNameConflicts(BindingDiagnosticBag diagnostics)
 
                     var conversion = symbol as SourceUserDefinedConversionSymbol;
                     var method = symbol as SourceMemberMethodSymbol;
-                    if (!(conversion is null))
+
+                    // We don't want to consider explicit interface implementations
+                    if (conversion is { MethodKind: MethodKind.Conversion })
```

Comment:
Currently, `ConversionSignatureComparer.Comparer` appears to only be used by `conversionsAsConversions` in this code path.

Let me know if it you think it's worth adding an assert that the given `SourceUserDefinedConversionSymbol`s have `MethodKind == MethodKind.Conversion`